File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Action/Animations/AnimateSlowAttackAction.ts
import AbstractGameAction, {ActionType} from "mods/ModTheSpire/Scripts/Action/AbstractGameAction";
import AbstractCreature from "mods/ModTheSpire/Scripts/Unit/AbstractCreature";
import Async from "Core/Async";

export class AnimateSlowAttackAction extends AbstractGameAction{
    constructor(public owner : AbstractCreature) {
        super();
        this.Type = ActionType.WAIT;
    }

    async Update(): Promise<boolean> {
        this.owner.PlaySlowAttack();
        await Async.Delay(500);
        return true;
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Action/Combat/AnimateSlowAttackAction.cs
using Cysharp.Threading.Tasks;
using ModTheSpire.Event;
using ModTheSpire.Unit;

namespace ModTheSpire.Effect.Other
{
    public class AnimateSlowAttackAction : IGameAction
    {
        public float Duration { get; set; }
        public ActionType Type { get; } = ActionType.WAIT;
        private AbstractCreature Owner;
        public AnimateSlowAttackAction(AbstractCreature owner)
        {
            Owner = owner;
        }
        
        public void Init()
        {
        }

        public async UniTask<bool> Update()
        {
            Owner.DisplayObject.PlaySlowAttack();
            await UniTask.Delay(500);
            return true;
        }
    }
}